cmake/sdlchecks.cmake

```diff
@@ -930,7 +930,7 @@ macro(CheckPTHREAD)
     # Run some tests
     set(ORIG_CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}")
     set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} ${PTHREAD_CFLAGS} ${PTHREAD_LDFLAGS}")
-    if(CMAKE_CROSSCOMPILING)
+    if(CMAKE_CROSSCOMPILING OR DARWIN)
       set(HAVE_PTHREADS 1)
     else()
       check_c_source_runs("
```

Comment:
Instead of this, does it work if you change this `check_c_source_runs` to `check_c_source_compiles`?

Reply:
It theoretically should, however it wasn't clear to me if any other platforms / build configurations relied on actually using `check_c_source_runs`. (As mentioned above, there may be better longer-term fixes than all of this custom Pthread-detection code, but I didn't want to risk breaking anything else, nor do I currently have the time to test larger changes on all supported platforms, alas.)

Reply:
> It theoretically should, however it wasn't clear to me if any other platforms / build configurations relied on actually using `check_c_source_runs`.

I don't think so: autotools equivalent of the check was `AC_TRY_LINK` and not `AC_TRY_RUN`

If someone can confirm that changing `check_c_source_runs` to `check_c_source_compiles`
fixes the issue, I will push such a patch. Otherwise, I'll apply this pull request.